File: pyg_parent/pyg_shop_web/src/main/webapp/js/service/typeTemplateService.js
app.service('typeTemplateService', function ($http) {



    this.findSpecList = function (typeId) {
        return $http.get('../typeTemplate/findSpecList/' + typeId)
    }

    //查询实体
    this.findOne = function (id) {
        return $http.get('../typeTemplate/findOne/' + id);
    }
    //增加
    this.add = function (entity) {
        return $http.post('../typeTemplate/add', entity);
    }
    //修改
    this.update = function (entity) {
        return $http.post('../typeTemplate/update', entity);
    }
    //删除
    this.dele = function (ids) {
        return $http.get('../typeTemplate/delete/' + ids);
    }
    //搜索
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../typeTemplate/search/' + pageNum + "/" + pageSize, searchEntity);
    }
});

File: pyg_parent/pyg_manager/pyg_manager_interface/src/main/java/com/pinyougou/manager/service/TypeTemplateService.java
package com.pinyougou.manager.service;
import java.util.List;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

import com.pinyougou.pojo.TbTypeTemplate;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface TypeTemplateService {

	
	/**
	 * 增加
	*/
	public void add(TbTypeTemplate typeTemplate);
	
	
	/**
	 * 修改
	 */
	public void update(TbTypeTemplate typeTemplate);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbTypeTemplate findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbTypeTemplate typeTemplate, int pageNum, int pageSize);

	/**
	 * 查询所有模板
	 * @return
	 */
    List<Map> findAll();

    List<Map> findSpecList(Long id);
}
